requests/exceptions.py

```diff
@@ -104,9 +104,18 @@ class StreamConsumedError(RequestException, TypeError):
 class RetryError(RequestException):
     """Custom retries logic failed"""
 
+
 class UnrewindableBodyError(RequestException):
     """Requests encountered an error when trying to rewind a body"""
 
+
+class ConflictingHeaderError(RequestException):
```

Comment:
Only minor note I have here is we've added an `InvalidHeader` exception since this PR was created which could likely be used instead of this. I didn't want to tweak the original work too much in the initial pass though. Thoughts?